### PR TITLE
Fix Map array deserialisation in Newtonsoft serialiser

### LINKED ISSUE #215: GraphQLResponse.Extensions not deserialised properly
I'm trying to include Apollo Tracing data in my responses, which gets included in the Extensions field of the GraphQLResponse object.

For example my query result returns data like in the apollo example at https://github.com/apollographql/apollo-tracing#example

When I look at the values in the extensions map, it doesn't seem to have deserialised the trace extension object correctly.

`extensions.tracing.parsing` and similar fields are fine, but `extensions.tracing.execution.resolvers` is always an empty array.

This seems to be a problem in `GraphQL.Client.Serializer.Newtonsoft.MapConverter.ReadArray`, because the items in the resolvers array are of type `JTokenType.Property` and are therefore skipped / ignored.

Is it possible to extend the deserialiser to support this structure of extension JSON?

### PR DESCRIPTION
**Pull request by PaulKlein**
Resolves #215

### CODE REVIEW

**Comment by PaulKlein** [comment]
Sorry, for some reason I got the issue number way wrong when committing this but I'm not sure if I can change it.

I am also not sure what's causing the failing test - any ideas?

**Comment by rose-a** [comment]
Did the tests pass locally? There is a persistent issue with the integration tests where some tests randomly fail on GitHub Actions...

File: src/GraphQL.Client.Serializer.Newtonsoft/MapConverter.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ token switch
 
         private IEnumerable<object> ReadArray(JToken element)
```
Comment (rose-a) [inline src/GraphQL.Client.Serializer.Newtonsoft/MapConverter.cs:54]:
```suggestion
        private IEnumerable<object> ReadArray(JArray element)
```

File: src/GraphQL.Client.Serializer.Newtonsoft/MapConverter.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ token switch
...
         {
-            foreach (var item in element.Values())
+            foreach (var item in (JArray)element)
```
Comment (rose-a) [inline src/GraphQL.Client.Serializer.Newtonsoft/MapConverter.cs:56]:
```suggestion
            foreach (var item in element)
```

**Comment by rose-a** [comment]
Tests pass on my development machine, so nothing in the way of merging your modifications... 

Thanks for digging into this!

**Comment by PaulKlein** [comment]
Thanks for taking a look at this so quickly.

Yes, your changes tidy this up even more, thanks for the feedback there.
The tests do run fine on my machine as well

Thanks again

**Comment by rose-a** [comment]
Included in v3.0.3 😉

**Pull request merged**